build/vaultwarden/build.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/bash
+#
+# {{{ CDDL HEADER
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+# }}}
+
+# Copyright 2022 OmniOS Community Edition (OmniOSce) Association.
+
+. ../../lib/build.sh
+
+PROG=vaultwarden
+VER=1.24.0
+PKG=ooce/server/vaultwarden
+SUMMARY="Bitwarden compatible server"
+DESC="Unofficial Bitwarden compatible server written in Rust, formerly known "
+DESC+="as bitwarden_rs"
+
+WEBVAULTVER=2.27.0
+
+RUST=rust-nightly-x86_64-unknown-illumos
+RUSTVER=2022-01-23
+
+set_arch 64
+
+BASEDIR=$PREFIX/$PROG
+CONFFILE=/var$BASEDIR/env.template
+WEBVAULTDIR=/var$BASEDIR/web-vault
+EXECFILE=$PREFIX/bin/$PROG
+
+BMI_EXPECTED=1
+CARGO_ARGS="--features sqlite,mysql,postgresql"
+BUILD_DEPENDS_IPS="
+    database/sqlite-3
+    ooce/library/mariadb-${MARIASQLVER//./}
+    ooce/library/postgresql-${PGSQLVER//./}
+"
+
+
+get_rust_nightly() {
+    BUILDDIR=$RUST download_source rust/nightly/$RUSTVER $RUST
+
+    logmsg "Installing rust nightly [$RUSTVER]"
+    logcmd $TMPDIR/$RUST/install.sh \
+        --prefix=$TMPDIR/_rust || logerr "installing rust"
+
+    CARGO="$TMPDIR/_rust/bin/cargo"
+    export PATH="$TMPDIR/_rust/bin:$PATH"
+}
+
+copy_sample_config() {
+    local relative_conffile=${CONFFILE#/}
+    local dest_confdir=$DESTDIR/${relative_conffile%/*}
+
+    logmsg "-- copying sample config"
+    logcmd mkdir -p "$dest_confdir" || logerr "mkdir failed"
+    logcmd cp $TMPDIR/$BUILDDIR/.env.template $DESTDIR/$relative_conffile \
+        || logerr "copying configs failed"
+}
+
+get_webvault() {
+    local danigarcia=$GITHUB/dani-garcia
+    local prog_repo=bw_web_builds
+    local prog=web-vault
+    local relative_webvaultdir=${WEBVAULTDIR#/}
+    local dest_webvaultdir=$DESTDIR/${relative_webvaultdir%/*}
+
+    note -n "Pulling v$WEBVAULTVER prebuilt $prog"
+
+    save_variables SKIP_CHECKSUM
```

Comment:
Is it a good idea to do this save/restore variable dance in this function?

Reply:
It can probably be dropped. We can just set `SKIP_CHECKSUM=1` for the download call itself.
Since the web stuff is downloaded last, I think overwriting the variables is fine, and we just need to check that the `info.source-url` entries in the final manifest look ok.